### PR TITLE
RWA-934: Set-up a Kubernetes CronJob to log errors for alerts

### PR DESCRIPTION
**Pull request by mkayad**
### JIRA link (if applicable) ###

[RWA-934](https://tools.hmcts.net/jira/browse/RWA-934)

### Change description ###

See ticket

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: k8s/aat/cluster-00-overlay/wa/kustomization.yaml [hunk 1]
```diff
@@ -6,8 +6,10 @@ bases:
   - ../../../namespaces/wa/wa-task-batch-initiation/wa-task-batch-initiation.yaml
   - ../../../namespaces/wa/wa-task-batch-termination/wa-task-batch-termination.yaml
   - ../../../namespaces/wa/wa-task-batch-ad-hoc-pending-termination/wa-task-batch-ad-hoc-pending-termination.yaml
+  - ../../../namespaces/wa/wa-message-find-problem-messages/wa-message-find-problem-messages.yaml
```
Comment (raghera) [inline k8s/aat/cluster-00-overlay/wa/kustomization.yaml:9]: I would pluralise the prefix we are using i.e. `wa-messages-. . .`

File: k8s/namespaces/wa/wa-message-find-problem-messages/aat-00.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+apiVersion: helm.fluxcd.io/v1
+kind: HelmRelease
+metadata:
+  name: wa-message-find-problem-messages
+spec:
+  values:
+    job:
+      schedule: "0,4,8,12,16,20,24,28,32,36,40,44,48,52,56 * * * *"
```
Comment (raghera) [inline k8s/namespaces/wa/wa-message-find-problem-messages/aat-00.yaml:8]: Could make it less frequent to start like every 15 mins?  I.e. to see if it is working ok.

**Review by raghera — COMMENTED** [review]
I think there are some other files to update like `image-policy.yaml` and then you need to refer to that in `automation/kustomization.yaml`

You can check by find everywhere where you find `batch-initiation` or any other job we know works.

File: k8s/aat/cluster-00-overlay/wa/kustomization.yaml (hunk 1 above)
Comment (mkayad) [reply k8s/aat/cluster-00-overlay/wa/kustomization.yaml:9]: done

File: k8s/namespaces/wa/wa-message-find-problem-messages/aat-00.yaml (hunk 2 above)
Comment (mkayad) [reply k8s/namespaces/wa/wa-message-find-problem-messages/aat-00.yaml:8]: Done

File: k8s/aat/cluster-00-overlay/wa/kustomization.yaml [hunk 3]
```diff
@@ -6,8 +6,10 @@ bases:
   - ../../../namespaces/wa/wa-task-batch-initiation/wa-task-batch-initiation.yaml
   - ../../../namespaces/wa/wa-task-batch-termination/wa-task-batch-termination.yaml
   - ../../../namespaces/wa/wa-task-batch-ad-hoc-pending-termination/wa-task-batch-ad-hoc-pending-termination.yaml
+  - ../../../namespaces/wa/wa-messages-find-problem-messages/wa-messages-find-problem-messages.yaml
 patchesStrategicMerge:
   - ../../../namespaces/wa/wa-task-batch-configuration/aat-00.yaml
   - ../../../namespaces/wa/wa-task-batch-initiation/aat-00.yaml
   - ../../../namespaces/wa/wa-task-batch-termination/aat-00.yaml
   - ../../../namespaces/wa/wa-task-batch-ad-hoc-pending-termination/aat-00.yaml
+  - ../../../namespaces/wa/wa-messages-find-problem-messages/aat-00.yaml
```
Comment (raghera) [inline k8s/aat/cluster-00-overlay/wa/kustomization.yaml:15]: should be running in both clusters I think.

File: k8s/namespaces/wa/wa-messages-find-problem-messages/aat-00.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+apiVersion: helm.fluxcd.io/v1
+kind: HelmRelease
+metadata:
+  name: wa-messages-find-problem-messages
+spec:
+  values:
+    job:
+      schedule: "0,30* * * *"
```
Comment (raghera) [inline k8s/namespaces/wa/wa-messages-find-problem-messages/aat-00.yaml:8]: Think you need a space between, otherwise it's invalid

Comment (mkayad) [reply k8s/namespaces/wa/wa-messages-find-problem-messages/aat-00.yaml:8]: got it

File: apps/wa/automation/kustomization.yaml [hunk 5]
```diff
@@ -26,3 +26,4 @@ resources:
   - ../wa-task-management-api/image-policy.yaml
   - ../wa-workflow-api/image-repo.yaml
   - ../wa-workflow-api/image-policy.yaml
+  - ../wa-messages-find-problem-messages/image-policy.yaml
```
[the remark this answers is not in the document]
Comment (mkayad) [reply apps/wa/automation/kustomization.yaml:30]: Done

**Review by timja — APPROVED** [review]

**Pull request merged**